tests/zfs-tests/tests/functional/casenorm/mixed_create_failure.ksh

```diff
@@ -0,0 +1,56 @@
+#!/bin/ksh -p
+#
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+#
+# Copyright 2018 Nutanix, Inc.  All rights reserved.
+#
+
+. $STF_SUITE/tests/functional/casenorm/casenorm.kshlib
+
+# DESCRIPTION:
+# For the filesystem with casesensitivity=mixed, normalization=none,
+# when multiple files which the same name (differing only in case) created, the
+# number of files is limited to what can fit in a fatzap leaf-block. And beyond
+# that, it fails with ENOSPC. Ensure that the create fails gracefully and does
+#  not hit and assert and panic.
+#
+# STRATEGY:
+# 1. Create files with same name but varying in case.
+#    E.g. 'abcdefghijklmnop', 'Abcdefghijklmnop', 'ABcdefghijklmnop' etc.
+#    The create would fail with ENOSPC.
```

Comment:
Since this change touches several system calls we should extent the test case to verify they behave as expected too.  Specifically let's check rename, symlinks, and mkdir.